Docs/Benchmarks.md

```diff
@@ -5,45 +5,101 @@
+| Alveo U280 | ParallelAlgorithm                 | 226%            | 1858%           | 300         | 362 ms    | 33 Ws     | 10.86%           | 102 ms   | 111 ms     | 1.66 Ws    | 14.99 W            |
+| Alveo U280 | MonteCarloPiEstimator             | 387%            | 2397%           | 230         | 185 ms    | 17 Ws     | 13.63%           | 16 ms    | 38 ms      | 0.67 Ws    | 17.55 W            |
+| Alveo U250 | ImageContrastModifier<sup>1</sup> | 1503%           | 5621%           | 150         | 529 ms    | 48 Ws     | 18.29%           | 13 ms    | 33 ms      | 0.83 Ws    | 25.22 W            |
+| Alveo U250 | ImageContrastModifier<sup>2</sup> | 3268%           | 11921%          | 150         | 193158 ms | 17384 Ws  | 18.29%           | 5535 ms  | 5735 ms    | 144.61 Ws  | 25.22 W            |
+| Alveo U250 | ParallelAlgorithm                 | 357%            | 2437%           | 300         | 498 ms    | 45 Ws     | 10.30%           | 101 ms   | 109 ms     | 1.77 Ws    | 16.21 W            |
+| Alveo U250 | MonteCarloPiEstimator             | 369%            | 2022%           | 230         | 197 ms    | 18 Ws     | 12.39%           | 21 ms    | 42 ms      | 0.84 Ws    | 19.89 W            |
+| Alveo U200 | ImageContrastModifier<sup>1</sup> | 735%            | 3047%           | 150         | 543 ms    | 49 Ws     | 27.23%           | 12 ms    | 65 ms      | 1.55 Ws    | 23.89 W            |
+| Alveo U200 | ImageContrastModifier<sup>2</sup> | 3448%           | 13266%          | 150         | 198172 ms | 17835 Ws  | 27.23%           | 5340 ms  | 5586 ms    | 133.44 Ws  | 23.89 W            |
+| Alveo U200 | ParallelAlgorithm                 | 171%            | 1464%           | 300         | 379 ms    | 34 Ws     | 15.56%           | 110 ms   | 140 ms     | 2.18 Ws    | 15.58 W            |
+| Alveo U200 | MonteCarloPiEstimator             | 203%            | 1333%           | 230         | 203 ms    | 18 Ws     | 18.57%           | 17 ms    | 67 ms      | 1.28 Ws    | 19.04 W            |
+| Alveo U50  | ImageContrastModifier<sup>1</sup> | 1324%           | 6359%           | 150         | 470 ms    | 42 Ws     | 32.09%           | 12 ms    | 33 ms      | 0.65 Ws    | 19.85 W            |
+| Alveo U50  | ImageContrastModifier<sup>3</sup> | 3462%           | 16052%          | 150         | 17167 ms  | 1545 Ws   | 32.09%           | 450 ms   | 482 ms     | 9.57 Ws    | 19.85 W            |
+| Alveo U50  | ParallelAlgorithm                 | 258%            | 2653%           | 300         | 379 ms    | 34 Ws     | 16.22%           | 104 ms   | 106 ms     | 1.24 Ws    | 11.69 W            |
+| Alveo U50  | MonteCarloPiEstimator             | 348%            | 2693%           | 230         | 197 ms    | 18 Ws     | 20.37%           | 18 ms    | 44 ms      | 0.63 Ws    | 14.43 W            |

```

Comment:
Próbaképp lefuttattam ezt most egy U50 x16-os gépen, és nekem 15ms (14,9946) net, 33ms (minimum) total jöt ki Debug mode-ban. Az FPGA az determinisztikus ugye és mindig pontosan ugyanaz jön ki, szóval itt valami nem kerek. Az odáig OK, hogy nem lesz us-ra ugyanannyi, mert van kis eltérés a hardverben, de nálad 20%-ka nagyobb futásidő jött ki, ami ezzel nem magyarázható (vagy nagyon tré a Xilinx gyártása).

Nézd meg kérlek, hogy mi lehet az oka. Íme az egész mappa, szoftver (előre beconfigolva, szal csak `dotnet Hast.Samples.Consumer.dll`) és benna  HWF mappa is.
[Hastlayer.zip](https://github.com/Lombiq/Hastlayer-SDK/files/5613152/Hastlayer.zip)

CPU-n amúgy kb. 300ms, de ez nem túl meglepő a Debug mode miatt.

Reply:
Ezzel a binaryval nekem is ugyanazok a számok jöttek ki mint neked. A ciklus szám nagyjából megegyezik (itt 4393400, az benchmark verzióban pedig 4393419), viszont a frekvencia nagyon eltér (itt 293 MHz, ott csak 241 MHz volt) így a kapott ms is eltér. Ennek szerintem az az oka, hogy a Wigneres cluster1-en ahol a build történt 2019.2 verzió van. Gondolom optimalizálták a compilerüket a 2020.1-es platformokkal így nagyobb frekvencián is tud futni. Eszembe se jutott, hogy egyáltalán van különbség nemhogy ilyen nagy!

Nem meglepően az áramfogyasztás egy kicsit magasabb, 16.784 W on-chip power, de a kisebb totál itő miatt az összfogyasztás javul. Érdekes módon az FPGA Utilization viszont egy nagyon kicsit rosszabb, 20.37% helyett 20.35%. Azt hinné az ember itt lehet nagyot javítani de úgy látszik nem.

Kiegészítem a doksit:
> - The kernels were built on the initial 2019.2 version of the Vitis Unified Software Platform. We have seen in one case 20% improvement in frequency (leading to shorter run times and lower total power consumption) by compiling with the newer 2020.1 version.

Reply:
Aha, hát ez poén. OK, egyelőre jó lesz, aztán majd a 2020.2-vel újrapróbálkozhatunk.